discovery/azure/azure.go

```diff
@@ -115,6 +127,15 @@ func validateAuthParam(param, name string) error {
 	return nil
 }
 
+func setCacheParam(param model.Duration, name string) (*cache.Cache[string, *network.Interface], error) {
+	if param == 0 {
+		return nil, nil
+	}
+
+	l := cache.New(cache.AsLRU[string, *network.Interface](lru.WithCapacity(10000)))
```

Comment:
10000 seems a lot ? is this preallocated? do we need to preallocate?

Reply:
Yes this is pre allocated, I will decrease to 5000 that should be fine for most of use case.

Reply:
set to 5000